### PR TITLE
PCP-14: Improve TiKV metrics performance

### PR DESCRIPTION
**Pull request by Renkai**
Signed-off-by: Renkai <gaelookair@gmail.com>

###  What have you changed?
It  is  minimal change to show how to improve metrics performance for further discussion.
* Did I put `start` and `stop` of `coarsetime::Updater` at the right place ?
* Should https://github.com/tikv/tikv/compare/master...Renkai:batch_metrics#diff-79ba1b5dd0906a82a30f8ab9eba6ebd2R2460 be configurable somewhere?If yes, where should it be?
* Should I make https://github.com/tikv/tikv/compare/master...Renkai:batch_metrics#diff-79ba1b5dd0906a82a30f8ab9eba6ebd2R2457 a more abstract function?If yes,what a function signature should it have?

###  What is the type of the changes?

- Improvement (a change which is an improvement to an existing feature)

###  How is the PR tested?
- Unit test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No 
###  Does this PR affect `tidb-ansible`?

No 
###  Refer to a related PR or issue link (optional)
https://github.com/tikv/tikv/issues/5708

### CODE REVIEW

File: src/raftstore/store/fsm/apply.rs [hunk 1]
```diff
@@ -2448,8 +2449,22 @@ impl ApplyFsm {
                 self.delegate.pending_cmds.append_normal(cmd);
             }
         }
-        // TODO: observe it in batch.
-        APPLY_PROPOSAL.observe(propose_num as f64);
+
+        APPLY_PROPOSAL_LOCAL.with(|hist| {
```
Comment (siddontang) [inline src/raftstore/store/fsm/apply.rs]: @BusyJay 

maybe it is better to let these thread pool support OnTick mechanism?

Comment (Renkai) [reply src/raftstore/store/fsm/apply.rs]: @siddontang A OnTick mechanism implemented,please see the newest code

**Comment by siddontang** [comment]
PTAL @breeswish

**Comment by breeswish** [comment]
Hi @Renkai , since https://github.com/tikv/tikv/pull/6134 is merged, you may consider using the newly introduced utility `TLSMetricGroup` to help you build static metrics more cleanly.

**Comment by Renkai** [comment]
new pr at https://github.com/tikv/tikv/pull/6163

**Pull request closed without merging**